### PR TITLE
Item manager feature final

### PR DESCRIPTION
**Pull request by chenbriann**
This PR contains the final implementation of the item manager feature for the inventory program. Major changes were with regards to clean architecture in the code. Classes have been packaged into their correct layers, and dependencies have been changed to follow CA principles. In terms of features, the use cases for adding, removing and updating item quantity have been implemented. The ability to get the history of an item has also been implemented. UI has been created to display items in the inventory, as well as a menu when an item is clicked that gives the user the ability to update the item quantity and remove the item from the inventory. A button has been added at the bottom of the inventory UI to add new items to the inventory. Items are tracked via barcode, and a csv storing the barcodes is written to and read from in real time when adding and removing items from inventory.

### CODE REVIEW

**Review by BENEDIDIN — APPROVED** [review]
All these changes look good. You have a good adherence to clean architecture. I also like all the cleaning up you did of the warnings.

**Review by PeterDushniku — APPROVED** [review]
Good work Brian.

You did an excellent job at implementing clean architecture principles. You used controllers, exporters, etc. in a way that aligns with clean architectures and keeps the layers of the program separate.

Your tests also look good and test your code quite thoroughly.

I also like how you had different use case packages for each use case, as it keeps the code clean and also follows clean architecture & SOLID principles.

Also your UI looks quite good an feels intuitive. Being able to click on items in the table to access remove, item history, etc. is well done and a good design decision.

**Review by BENEDIDIN — APPROVED** [review]
All these changes look good. You have a good adherence to clean architecture. I also like all the cleaning up you did of the warnings. I think using files to store barcodes was a poor design choice. It seems to mess things up with and is unnecessary. I don't think there is any need for data persistence here. I think the barcodes could just be stored in an arrayList.

File: src/main/java/entities/Order.java [hunk 1]
```diff
@@ -7,6 +7,21 @@ public class Order extends Item{
     private final String supplier;
     private final int cases;
```
Comment (chenbriann) [inline src/main/java/entities/Order.java:9]: Added facade design pattern to item class for reading and generating barcodes to delegate barcode tasks to other classes.

**Review by bwwaaa — APPROVED** [review]
Looks great! Your appliance of clean architecture and well documented java docs and comments throughout your code is excellent, making reading and understanding your work easy to do. 
The different layers of clean architecture are apparent in your code and are easy to recognize thanks to your organization of classes through appropriate packages.
Variables are also consistent with the camel case type of naming. 
Great work!

**Pull request merged**